File: Kotlin/my-calendar-i.kt
class MyCalendar() {

    val events = arrayListOf<Pair<Int,Int>>()

    fun book(start: Int, end: Int): Boolean {
        for ((eventStart, eventEnd) in events) {
            if ((start < eventEnd) && (eventStart < end)) {
                return false
            }
        }
        events.add(Pair(start, end))
        //println(events)
        return true
    }

}

/**
 * Your MyCalendar object will be instantiated and called as such:
 * var obj = MyCalendar()
 * var param_1 = obj.book(start,end)
 */

File: Go/my-calendar-i.go
type MyCalendar struct {
	events [][]int
}

func Constructor() MyCalendar {
	return MyCalendar{}
}

func (this *MyCalendar) Book(start int, end int) bool {
	for _, event := range this.events {
		startEvent, endEvent := event[0], event[1]
		if start < endEvent && startEvent < end {
			return false
		}
	}
	this.events = append(this.events, []int{start, end})
	return true
}

/**
 * Your MyCalendar object will be instantiated and called as such:
 * obj := Constructor();
 * param_1 := obj.Book(start,end);
 */